src/governance-validators.cpp

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) 2014-2017 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "governance-validators.h"
+
+#include "base58.h"
+#include "utilstrencodings.h"
+
+#include <algorithm>
+
+CProposalValidator::CProposalValidator(const std::string& strDataHexIn)
+    : strData(),
+      objJSON(UniValue::VOBJ),
+      fJSONValid(false),
+      strErrorMessages()
+{
+    if(!strDataHexIn.empty()) {
+        SetHexData(strDataHexIn);
+    }
+}
+
+void CProposalValidator::Clear()
+{
+    strData = std::string();
+    objJSON = UniValue(UniValue::VOBJ);
+    fJSONValid = false;
+    strErrorMessages = std::string();
+}
+
+void CProposalValidator::SetHexData(const std::string& strDataHexIn)
+{
+    std::vector<unsigned char> v = ParseHex(strDataHexIn);
+    strData = std::string(v.begin(), v.end());
+    ParseJSONData();
+}
+
+bool CProposalValidator::Validate()
+{
+    if(!ValidateJSON()) {
+        return false;
```

Comment:
Those error messages are added by ParseJSONData. The only case not covered is if the data string is empty.  I will add a message for that.